pkg/controller/awsfederatedaccountaccess/awsfederatedaccountaccess_controller.go

```diff
@@ -768,3 +750,45 @@ func hasLabel(awsFederatedAccountAccess *awsv1alpha1.AWSFederatedAccountAccess,
 	}
 	return false
 }
+
+func checkAndDeletePolicy(reqLogger logr.Logger, awsClient awsclient.Client, uidLabel string, crPolicyName string, policyName *string, policyArn *string) error {
+
+	var awsCustomPolicyname string
+	awsCustomPolicyname = checkUidLabel(awsCustomPolicyname, crPolicyName, uidLabel)
+
+	if *policyName == awsCustomPolicyname {
+		_, err := awsClient.DeletePolicy(&iam.DeletePolicyInput{PolicyArn: policyArn})
+		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				switch aerr.Code() {
+				case iam.ErrCodeNoSuchEntityException:
```

Comment:
Are these new cases not a change in functionality? Here we'll be logging at info level and then step out of the block to `return nil`. You'll need to add `return err` to each case

Reply:
Why are these errors being logged at info level instead of error level?